datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/auth/CreateAccessTokenResolver.java

```diff
@@ -0,0 +1,106 @@
+package com.linkedin.datahub.graphql.resolvers.auth;
+
+import com.datahub.authentication.Actor;
+import com.datahub.authentication.ActorType;
+import com.datahub.authentication.token.StatefulTokenService;
+import com.datahub.authentication.token.TokenType;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.authorization.AuthorizationUtils;
+import com.linkedin.datahub.graphql.exception.AuthorizationException;
+import com.linkedin.datahub.graphql.generated.AccessToken;
+import com.linkedin.datahub.graphql.generated.AccessTokenType;
+import com.linkedin.datahub.graphql.generated.CreateAccessTokenInput;
+import com.linkedin.datahub.graphql.generated.EntityType;
+import com.linkedin.datahub.graphql.generated.AccessTokenMetadata;
+import com.linkedin.metadata.Constants;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.Date;
+import java.util.concurrent.CompletableFuture;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.*;
+
+
+/**
+ * Resolver for generating personal & service principal access tokens
+ */
+@Slf4j
+public class CreateAccessTokenResolver implements DataFetcher<CompletableFuture<AccessToken>> {
+
+  private final StatefulTokenService _statefulTokenService;
+
+  public CreateAccessTokenResolver(final StatefulTokenService statefulTokenService) {
+    _statefulTokenService = statefulTokenService;
+  }
+
+  @Override
+  public CompletableFuture<AccessToken> get(final DataFetchingEnvironment environment) throws Exception {
+    return CompletableFuture.supplyAsync(() -> {
+      final QueryContext context = environment.getContext();
+      final CreateAccessTokenInput input = bindArgument(environment.getArgument("input"), CreateAccessTokenInput.class);
+
+      log.info("User {} requesting new access token for user {} ", context.getActorUrn(), input.getActorUrn());
+
+      if (isAuthorizedToGenerateToken(context, input)) {
+        final TokenType type = TokenType.valueOf(input.getType().toString()); // warn: if we are out of sync with AccessTokenType there are problems.
+        final String actorUrn = input.getActorUrn();
+        final Date date = new Date();
+        final long createdAtInMs = date.getTime();
+        final long expiresInMs = AccessTokenUtil.mapDurationToMs(input.getDuration());
+
+        final String tokenName = input.getName() != null ? input.getName() : "";
+        final String tokenDescription = input.getDescription() != null ? input.getDescription() : "";
```

Comment:
No need to set to empty string. Better to just leave it null if its not provided.